test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,505 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_session]: block_session.c
+ * The block manager extent list consists of both extent blocks and size blocks. This unit test
+ * suite tests aims to test all of the allocation and frees of the extent and size block functions.
+ *
+ * The block session manages an internal caching mechanism for both block and size blocks that are
+ * created or discarded.
+ */
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include "../wrappers/mock_session.h"
+
+void
+validate_and_cleanup_ext_block(WT_EXT *ext)
+{
+    REQUIRE(ext != nullptr);
+    REQUIRE(ext->depth != 0);
+    __wt_free(nullptr, ext);
+}
+
+void
+validate_and_cleanup_size_block(WT_SIZE *size)
+{
+    REQUIRE(size != nullptr);
+    __wt_free(nullptr, size);
+}
+
+void
+cleanup_ext_list(WT_BLOCK_MGR_SESSION *bms)
+{
+    WT_EXT *curr = bms->ext_cache;
+    WT_EXT *tmp;
+    while (curr != NULL) {
+        tmp = curr;
+        curr = curr->next[0];
+        __wt_free(nullptr, tmp);
+    }
+}
+
+void
+validate_and_cleanup_ext_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    int i = 0;
+
+    REQUIRE(bms != nullptr);
+    WT_EXT *curr = bms->ext_cache;
+
+    REQUIRE(bms->ext_cache_cnt == expected_items);
+    for (; i < expected_items; i++) {
+        REQUIRE(bms->ext_cache != nullptr);
+        curr = curr->next[0];
+    }
+    REQUIRE(curr == nullptr);
+    cleanup_ext_list(bms);
+}
+
+void
+cleanup_size_list(WT_BLOCK_MGR_SESSION *bms)
+{
+    WT_SIZE *curr = bms->sz_cache;
+    WT_SIZE *tmp;
+    while (curr != NULL) {
+        tmp = curr;
+        curr = curr->next[0];
+        __wt_free(nullptr, tmp);
+    }
+}
+
+void
+validate_and_cleanup_size_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    int i = 0;
+
+    REQUIRE(bms != nullptr);
+    WT_SIZE *curr = bms->sz_cache;
+
+    REQUIRE(bms->sz_cache_cnt == expected_items);
```

Comment:
This also needs an additional test that if `sz_cache_cnt` is 0 then `sz_cache` is NULL.

Reply:
done.